src/main/java/cam72cam/immersiverailroading/entity/LocomotiveDiesel.java

```diff
@@ -156,16 +234,57 @@ public void onUpdate() {
 			burnTime *= getDefinition().getFuelEfficiency()/100f;
 			burnTime *= (Config.ConfigBalance.locoDieselFuelEfficiency / 100f);
 			
-			while (internalBurn < 0 && this.getLiquidAmount() > 0) {
+			/*while (internalBurn < 0 && this.getLiquidAmount() > 0) {
 				internalBurn += burnTime;
 				theTank.drain(1, true);
+			}*/
+			
+			float heatUpSpeed = 0.0029167f * Config.ConfigBalance.dieselLocoHeatTimeScale;
+			
+			if (getTurnedOn()) {
+				if (getEngineTemperature() < 73 && this.getLiquidAmount() > 0) {
+					setEngineTemperature(getEngineTemperature() + heatUpSpeed);
+					theTank.drain(1, true);
```

Comment:
This breaks the mechanic where a single mb of fuel might last more than 1 tick.  That's why the while loop you removed existed